app/components/SignInModal/index.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+const ModalCloseX = styled.span`
+  color: #ddd;
+  font-size: 3.25rem;
+  text-shadow: none;
+`;
+
+const SignInModalDialog = styled.div.attrs({
+  className: 'modal-dialog',
+})``;
+
+const SignInModalContent = styled.div.attrs({
+  className: 'modal-content',
+})`
+  &&& {
+    background: transparent;
+    border: transparent;
+
+    & > div {
+      border: inherit;
+    }
+  }
+`;
+
+const SignInModalBody = styled.div.attrs({
+  className: 'modal-body pt-2',
+})`
+  color: white;
+`;
+
+const StyledLabel = styled.label.attrs({
+  className: 'text-uppercase small mb-1',
+})``;
+
+const StyledInput = styled.input.attrs({
+  type: 'input',
+  className: 'form-control form-control-lg mb-2',
+})``;
+
+const StyledButton = styled.button.attrs({
+  type: 'button',
+  className: 'btn btn-primary btn-lg'
+})``;
+
+const SubmitButton = styled(StyledButton).attrs({
+  className: 'my-4',
+})`
+  width: 95%;
+  margin-left: 2.5%;
+  margin-right: 2.5%;
+`;
+
+const SocialSignInButton = styled(StyledButton).attrs({
+  className: 'col-5',
+})``;
+
+class SignInModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {value: ''};
+
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
We're not using this yet right? I think we can remove it for now

Reply:
Good thinking about state and stuff here. Definitely will come into play soon. Though I think we will handle it using redux, reducers & actions and stuff instead of internally. Will hopefully be more clear soon

Reply:
Yeah, I copied that over from the React site's setup and made it a "controlled component" as was suggested. But then I didn't use this stuff.